### PR TITLE
[SPARK-45694][SPARK-45695][SQL] Clean up deprecated API usage `View.force` and `ScalaNumberProxy.signum`

### PR DESCRIPTION
**Pull request by ivoson**
### What changes were proposed in this pull request?

Clean up deprecated API usage:
1. ScalaNumberProxy.signum -> use `sign` instead;
2. Map.view.mapValues.view.force.toMap -> replaced to Map.vew.mapValues.toMap;

### Why are the changes needed?
Eliminate compile warnings and no longer use deprecated scala APIs.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala [hunk 1]
```diff
@@ -362,10 +362,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]]
       case s: Seq[_] =>
         s.map(mapChild)
       case m: Map[_, _] =>
-        // `map.mapValues().view.force` return `Map` in Scala 2.12 but return `IndexedSeq` in Scala
-        // 2.13, call `toMap` method manually to compatible with Scala 2.12 and Scala 2.13
-        // `mapValues` is lazy and we need to force it to materialize
-        m.view.mapValues(mapChild).view.force.toMap
+        // `mapValues` is lazy and we need to force it to materialize by invoking `toIndexedSeq`
+        m.view.mapValues(mapChild).view.toIndexedSeq.toMap
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala:366]: How about just `m.view.mapValues(mapChild).toMap`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala [hunk 2]
```diff
@@ -784,13 +782,12 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]]
...
         Some(arg.asInstanceOf[BaseType].clone())
-      // `map.mapValues().view.force` return `Map` in Scala 2.12 but return `IndexedSeq` in Scala
-      // 2.13, call `toMap` method manually to compatible with Scala 2.12 and Scala 2.13
+      // `mapValues` is lazy and we need to force it to materialize by invoking `toIndexedSeq`
       case m: Map[_, _] => m.view.mapValues {
         case arg: TreeNode[_] if containsChild(arg) =>
           arg.asInstanceOf[BaseType].clone()
         case other => other
-      }.view.force.toMap // `mapValues` is lazy and we need to force it to materialize
+      }.view.toIndexedSeq.toMap
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala:790]: ditto

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala (hunk 1 above)
Comment (ivoson) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala]: Yeah, it looks better. Thanks, done.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala (hunk 2 above)
Comment (ivoson) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala]: done

**Review by LuciferYang — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala [hunk 3]
```diff
@@ -362,10 +362,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]]
       case s: Seq[_] =>
         s.map(mapChild)
       case m: Map[_, _] =>
-        // `map.mapValues().view.force` return `Map` in Scala 2.12 but return `IndexedSeq` in Scala
-        // 2.13, call `toMap` method manually to compatible with Scala 2.12 and Scala 2.13
-        // `mapValues` is lazy and we need to force it to materialize
-        m.view.mapValues(mapChild).view.force.toMap
+        // `mapValues` is lazy and we need to force it to materialize by converting to Map
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala:365]: Is this, `we need to force it to materialize`, correct with the new code?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (if these are all occurrences)

**Comment by dongjoon-hyun** [comment]
Could you re-trigger the failed test pipeline, @ivoson ?

**Comment by LuciferYang** [comment]
image

@ivoson Please update the pr description.

**Comment by ivoson** [comment]
> Could you re-trigger the failed test pipeline, @ivoson ?

Hi @dongjoon-hyun done, the test passed now.

**Comment by ivoson** [comment]
> image
> @ivoson Please update the pr description.

Thanks, @LuciferYang . done

**Comment by LuciferYang** [comment]
Merged into master for Spark 4.0. Thanks @ivoson and @dongjoon-hyun ~

**Pull request closed without merging**